modules/cudaarithm/src/cuda/threshold.cu

```diff
@@ -101,8 +101,9 @@ double cv::cuda::threshold(InputArray _src, OutputArray _dst, double thresh, dou
 
     const int depth = src.depth();
 
-    CV_Assert( src.channels() == 1 && depth <= CV_64F );
+    CV_Assert( depth <= CV_64F );
     CV_Assert( type <= 4 /*THRESH_TOZERO_INV*/ );
+    src = src.reshape(src.channels(), src.rows);
```

Comment:
The `reshape` call will leave `src` mat with the same number of channels as it was.

IMO, you need to call

```
src = src.reshape(1);
dst = dst.reshape(1);
```

after `dst` mat initialization.